daemon/main.c

```diff
@@ -1497,8 +1305,23 @@ int main(int argc, char **argv) {
     info("netdata initialization completed. Enjoy real-time performance monitoring!");
     netdata_ready = 1;
 
+    set_late_global_environment();
+
     send_statistics("START", "-",  "-");
 
+    //check if ANALYTICS needs to start
+    if (netdata_anonymous_statistics_enabled == 1) {
+        for (i = 0; static_threads[i].name != NULL; i++) {
+            if (!strncmp(static_threads[i].name, "ANALYTICS", 9)) {
+                struct netdata_static_thread *st = &static_threads[i];
+                st->thread = mallocz(sizeof(netdata_thread_t));
+                st->enabled = 1;
+                debug(D_SYSTEM, "Starting thread %s.", st->name);
+                netdata_thread_create(st->thread, st->name, NETDATA_THREAD_OPTION_DEFAULT, st->start_routine, st);
+            }
+        }
+    }
```

Comment:
What for do you need this part? `netdata_anonymous_statistics_enabled` is always -1.

Reply:
Hi Vlad! Well, `netdata_anonymous_statistics_enabled` starts initially as `0` (defined as a global variable, on top of main.c). At some point in main.c (https://github.com/MrZammler/netdata/blob/28e7ea2e4306fe53cb2d1cf8e517a2ec0609f2f2/daemon/main.c#L1260), it gets the value of `-1`, i.e. we don't know if we're gonna send analytics. The first call to `send_statistics` will decide if the value will be `1` (i.e. send) or `0` (dont send).

The point of this (and having the analytics thread as `enabled = 0` is to only run the thread if we are indeed going to send analytics.

Reply:
Oh, sorry, my bad! I didn't notice that it is changed in `netdata_anonymous_statistics_enabled`.

Then I have another question. Why did you add `NETDATA_PLUGIN_HOOK_ANALYTICS` to `static_threads`? It won't start anyway.

Reply:
Well, the first iterations of the PR always started the analytics thread... But, It's there also because the agent (if the thread eventually runs) will handle it the same way as other threads (send cancel thread, etc).